### PR TITLE
Feat/issue 976/ecdata create constants for call data sizes

### LINKED ISSUE #976: feat(ecdata): create constants for call data sizes
should be constant of the class (create them in the constraints and use them)

_Originally posted by @letypequividelespoubelles in https://github.com/Consensys/linea-tracer/pull/748#discussion_r1730797448_

### PR DESCRIPTION
**Pull request by lorenzogentile404**

### CODE REVIEW

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/ecdata/EcDataOperation.java [hunk 1]
```diff
@@ -141,11 +141,11 @@ private EcDataOperation(
...
         switch (precompileFlag) {
-          case PRC_ECRECOVER -> 128;
-          case PRC_ECADD -> 128;
-          case PRC_ECMUL -> 96;
+          case PRC_ECRECOVER -> TOTAL_SIZE_ECRECOVER_DATA;
+          case PRC_ECADD -> TOTAL_SIZE_ECADD_DATA;
+          case PRC_ECMUL -> TOTAL_SIZE_ECMUL_DATA;
           case PRC_ECPAIRING -> {
-            Preconditions.checkArgument(callDataSize % 192 == 0);
+            Preconditions.checkArgument(callDataSize % TOTAL_SIZE_ECPAIRING_DATA_MIN == 0);
```
Comment (OlivierBBB) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/ecdata/EcDataOperation.java:148]: The use of `MIN` isn't right. It's the size of a pair of points (small + large) for `ECPAIRING` and every invokation must have `call_data_size` that is a multiple of that. In particular it's neither a `TOTAL_SIZE` nor a `MIN` (empty call data is allowed, too, though it won't transition through the present module.)

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/ecdata/EcDataOperation.java [hunk 2]
```diff
@@ -141,11 +141,11 @@ private EcDataOperation(
         "EcDataOperation should only be called with nonempty call rightPaddedCallData");
     final int paddedCallDataLength =
         switch (precompileFlag) {
-          case PRC_ECRECOVER -> 128;
-          case PRC_ECADD -> 128;
-          case PRC_ECMUL -> 96;
+          case PRC_ECRECOVER -> TOTAL_SIZE_ECRECOVER_DATA;
```
Comment (OlivierBBB) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/ecdata/EcDataOperation.java:144]: Maybe
```rust
ECRECOVER_BYTE_SIZE_TO_PAD_AND_TRIM_TO
ECADD_BYTE_SIZE_TO_PAD_AND_TRIM_TO
ECMUL_BYTE_SIZE_TO_PAD_AND_TRIM_TO
```

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/ecdata/EcDataOperation.java (hunk 1 above)
Comment (OlivierBBB) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/ecdata/EcDataOperation.java:148]: Maybe
```rust
ECPAIRING_BYTE_SIZE_OF_PAIR_OF_POINTS
```

Comment (OlivierBBB) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/ecdata/EcDataOperation.java:148]: And obviously these constants would have to be declared in the LISP. The lisp modules that would make use of these constants would be
- HUB
- OOB
- ... maybe other modules, too ?

Comment (lorenzogentile404) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/ecdata/EcDataOperation.java:148]: they are already part of the lisp and the specs

**Review by letypequividelespoubelles — APPROVED** [review]
LGTM

**Pull request merged**